mimium-lang/src/compiler/typing.rs

```diff
@@ -12,6 +12,10 @@ use std::collections::BTreeMap;
 use std::fmt;
 use std::rc::Rc;
 
+// Stage type matching interpreter.rs
+const PERSISTENT_STAGE: i64 = i64::MIN;
+type Stage = i64;
```

Comment:
The same definition as intepreter.rs is redundant.

`plugin::EvalStage` is more appropriate for this purpose.
Move the definition of EvalStage to compiler.rs and replace Stage in typing.rs and intepreter.rs